modules/programs/firefox/mkFirefoxModule.nix

```diff
@@ -452,104 +399,31 @@ in {
           };
 
           bookmarks = mkOption {
-            internal = !enableBookmarks;
-            type = let
-              bookmarkSubmodule = types.submodule ({ config, name, ... }: {
-                options = {
-                  name = mkOption {
-                    type = types.str;
-                    default = name;
-                    description = "Bookmark name.";
-                  };
-
-                  tags = mkOption {
-                    type = types.listOf types.str;
-                    default = [ ];
-                    description = "Bookmark tags.";
-                  };
+            type = (with types;
+              coercedTo (listOf anything) (bookmarks:
```

Comment:
I used `(listOf anything)` here because I couldn't find a nice way to pull out the original type from the submodule.

I don't believe this is an issue since this will just redirect you to the new submodule format.